src/libs/auth.ts

```diff
@@ -109,20 +120,34 @@ const getSigninArgs = (includeBillingScope) => {
 
 export const signIn = async (includeBillingScope = false) => {
   const args = getSigninArgs(includeBillingScope);
-  const user = await getAuthInstance().signinPopup(args);
-
+  const user: User = await getAuthInstance().signinPopup(args);
+  const generatedUuid = uuid();
+  const sessionStartTime = Date.now();
+  authStore.update((state) => ({
+    ...state,
+    sessionId: generatedUuid,
+    sessionStartTime,
+  }));
   // For B2C record in the auth store whether we requested the GCP cloud-billing scope since there
   // is no way to determine it after the fact.
   // For Google we don't need to do this since we can inspect the scope directly in the user object.
   if (!isGoogleAuthority()) {
```

Comment:
It's occurring to me there is some leftover code to support the Google -> B2C migration. Specifically `isGoogleAuthority()` is _never_ true anymore.. we are fully on B2C (and have been for >6 months).

Would it be in scope of this PR to remove `isGoogleAuthority()` and related code? That would let you collapse the 2 `authStore.update` calls into 1 in this method.